### DISCUSSION TITLE
Scan From File

### DISCUSSION THREAD

**Discussion by Sam1285** [Authenticator-Extension/Authenticator#629] [General]
<You can also suggest feedback by Tweeting with #AuthenticatorFeedback>

**Describe the feature you want:**
How about scan QR Codes from files like PDFs or any other medium?

**Why do you want this feature in Authenticator?**
More flexibility and usability. Got some authentications just as PDFs.

**Comment by Sneezry** [comment]
I think it is out of scope to a brower extension. But I'm open for discussion.

**Comment by mymindstorm** [comment]
Specifically dealing with PDFs is not easy due to extensions not being able to interact with them in Firefox. Although, you can take a screenshot and scan the QR from that.

**Comment by MarkHostettler** [comment]
I have been using the addin for over a year and have wondered this several times.  I use several different authentication apps, all allow me to print or store the QR code.  If I want to add one account from another app it would be nice to be able to scan that QR code from a file, simply view the contents of the file using whatever will open the QR code file, then scan from the addin.  Right now the QR code has to be displayed in the browser page, but it would be nice if it could scan QR codes displayed in other apps.

**Comment by mymindstorm** [comment]
> nice if it could scan QR codes displayed in other apps.

Unfortunately, this is very hard to do due to the limitations of the browser extensions platform.

**Comment by MarkHostettler** [comment]
I can see how difficult that could be after thinking about it.  I have adopted the method of setting up the MFA in the addin first, then using the stored QR code to setup the mobile apps.  Just difficult to do the reverse.  Thanks for the response.
